cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hdfs/AbstractHDFSStats.java

```diff
@@ -39,4 +63,85 @@
   public String getServiceName() {
     return SERVICE_NAME;
   }
+
+  public String getWebURL() {
+    try {
+      if (HAUtil.isHAEnabled(conf, getNameService())) {
+        URL haWebURL = getHAWebURL();
+        if (haWebURL != null) {
+          return haWebURL.toString();
+        }
+      } else {
+        try (FileSystem fs = FileSystem.get(conf)) {
+          URL webUrl = rpcToHttpAddress(fs.getUri());
+          if (webUrl != null) {
+            return webUrl.toString();
+          }
+        }
+      }
+      lastCollectFailed = false;
+    } catch (Exception e) {
+      // TODO: remove once CDAP-7887 is fixed
+      if (!lastCollectFailed) {
```

Comment:
Use sampling logger instead. See https://github.com/caskdata/cdap/blob/develop/cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogLocation.java#L59 for example usage